<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ahmed-Amine Anzali, Institut Gustave Roussy, INSERM U1015, Ecole Polytechnique">

<title>TCGA Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="TCGA_Analysis_for_Paper_files/libs/clipboard/clipboard.min.js"></script>
<script src="TCGA_Analysis_for_Paper_files/libs/quarto-html/quarto.js"></script>
<script src="TCGA_Analysis_for_Paper_files/libs/quarto-html/popper.min.js"></script>
<script src="TCGA_Analysis_for_Paper_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="TCGA_Analysis_for_Paper_files/libs/quarto-html/anchor.min.js"></script>
<link href="TCGA_Analysis_for_Paper_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="TCGA_Analysis_for_Paper_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="TCGA_Analysis_for_Paper_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="TCGA_Analysis_for_Paper_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="TCGA_Analysis_for_Paper_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#importing-tcga-hnsc-data-with-tcga-biolinks" id="toc-importing-tcga-hnsc-data-with-tcga-biolinks" class="nav-link active" data-scroll-target="#importing-tcga-hnsc-data-with-tcga-biolinks"><span class="toc-section-number">1</span>  Importing TCGA HNSC data with TCGA biolinks</a>
  <ul class="collapse">
  <li><a href="#importing-rnaseq-data" id="toc-importing-rnaseq-data" class="nav-link" data-scroll-target="#importing-rnaseq-data"><span class="toc-section-number">1.1</span>  Importing RNAseq data</a></li>
  <li><a href="#transform-into-ensembl-gene-ids-using-biomart" id="toc-transform-into-ensembl-gene-ids-using-biomart" class="nav-link" data-scroll-target="#transform-into-ensembl-gene-ids-using-biomart"><span class="toc-section-number">1.2</span>  Transform into ENSEMBL gene IDs using biomaRt</a></li>
  </ul></li>
  <li><a href="#retrieving-clinical-data" id="toc-retrieving-clinical-data" class="nav-link" data-scroll-target="#retrieving-clinical-data"><span class="toc-section-number">2</span>  Retrieving Clinical Data</a></li>
  <li><a href="#deg-analysis-using-deseq2" id="toc-deg-analysis-using-deseq2" class="nav-link" data-scroll-target="#deg-analysis-using-deseq2"><span class="toc-section-number">3</span>  DEG analysis using DESEq2</a>
  <ul class="collapse">
  <li><a href="#running-deseq2-model" id="toc-running-deseq2-model" class="nav-link" data-scroll-target="#running-deseq2-model"><span class="toc-section-number">3.1</span>  Running DESEq2 model</a></li>
  <li><a href="#volcano-plot" id="toc-volcano-plot" class="nav-link" data-scroll-target="#volcano-plot"><span class="toc-section-number">3.2</span>  Volcano Plot</a></li>
  <li><a href="#ma-plot" id="toc-ma-plot" class="nav-link" data-scroll-target="#ma-plot"><span class="toc-section-number">3.3</span>  MA Plot</a></li>
  <li><a href="#bulk-rnaseq-signature-genes" id="toc-bulk-rnaseq-signature-genes" class="nav-link" data-scroll-target="#bulk-rnaseq-signature-genes"><span class="toc-section-number">3.4</span>  Bulk RNAseq signature genes</a></li>
  <li><a href="#common-gene-signature-with-visium-spots" id="toc-common-gene-signature-with-visium-spots" class="nav-link" data-scroll-target="#common-gene-signature-with-visium-spots"><span class="toc-section-number">3.5</span>  Common gene signature with Visium Spots</a></li>
  <li><a href="#gene-ontology-based-on-degs" id="toc-gene-ontology-based-on-degs" class="nav-link" data-scroll-target="#gene-ontology-based-on-degs"><span class="toc-section-number">3.6</span>  Gene Ontology based on DEGs</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="toc-section-number">4</span>  Session Info</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">TCGA Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Ahmed-Amine Anzali, Institut Gustave Roussy, INSERM U1015, Ecole Polytechnique </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="importing-tcga-hnsc-data-with-tcga-biolinks" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Importing TCGA HNSC data with TCGA biolinks</h1>
<section id="importing-rnaseq-data" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="importing-rnaseq-data"><span class="header-section-number">1.1</span> Importing RNAseq data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Query platform for HNSC bulk RNA sequencing data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>query <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">GDCquery</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">"TCGA-HNSC"</span>, </span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.category =</span> <span class="st">"Transcriptome Profiling"</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.type =</span> <span class="st">"Gene Expression Quantification"</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Download the data for HNSC patients</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>TCGAbiolinks<span class="sc">::</span><span class="fu">GDCdownload</span>(query,<span class="at">files.per.chunk =</span> <span class="dv">10</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare expression matrix with geneID in the rows and samples (barcode) in the columns</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># rsem.genes.results as values</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>HNSC.Rnaseq.SE <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">GDCprepare</span>(query)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co"># For gene expression if you need to see a boxplot correlation and AAIC plot to define outliers you can run</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>HNSC.RNAseq_CorOutliers <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">TCGAanalyze_Preprocessing</span>(HNSC.Rnaseq.SE)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co"># normalization of genes</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>dataNorm <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">TCGAanalyze_Normalization</span>(</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">tabDF =</span> HNSC.RNAseq_CorOutliers, </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">geneInfo =</span>  TCGAbiolinks<span class="sc">::</span>geneInfoHT</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="co"># quantile filter of genes</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>dataFilt <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">TCGAanalyze_Filtering</span>(</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">tabDF =</span> dataNorm,</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"quantile"</span>, </span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">qnt.cut =</span>  <span class="fl">0.25</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a><span class="co"># selection of normal samples "NT"</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>samplesNT <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">TCGAquery_SampleTypes</span>(</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">barcode =</span> <span class="fu">colnames</span>(dataFilt),</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">typesample =</span> <span class="fu">c</span>(<span class="st">"NT"</span>)</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="co"># selection of tumor samples "TP"</span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>samplesTP <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">TCGAquery_SampleTypes</span>(</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">barcode =</span> <span class="fu">colnames</span>(dataFilt), </span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">typesample =</span> <span class="fu">c</span>(<span class="st">"TP"</span>)</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>datasampleNT <span class="ot">&lt;-</span> dataFilt[,samplesNT]</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>datasampleTP <span class="ot">&lt;-</span> dataFilt[,samplesTP]</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="co"># write into csv</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(datasampleTP, <span class="st">'RNAseq/Tumor_HNSC_bulk.csv'</span>,<span class="at">row.names =</span> T)</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(datasampleNT, <span class="st">'RNAseq/Normal_HNSC_bulk.csv'</span>,<span class="at">row.names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="transform-into-ensembl-gene-ids-using-biomart" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="transform-into-ensembl-gene-ids-using-biomart"><span class="header-section-number">1.2</span> Transform into ENSEMBL gene IDs using biomaRt</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'RNAseq/Tumor_HNSC_bulk.csv'</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rna) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">d+$"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(rna))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>ensembl.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(rna)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>geneIDs1 <span class="ot">&lt;-</span> ensembldb<span class="sc">::</span><span class="fu">select</span>(EnsDb.Hsapiens.v86<span class="sc">::</span>EnsDb.Hsapiens.v86, <span class="at">keys=</span> ensembl.genes, <span class="at">keytype =</span> <span class="st">"GENEID"</span>, <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>,<span class="st">"GENEID"</span>))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep Only Unique ID Names</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>geneIDs1 <span class="ot">&lt;-</span> geneIDs1 <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(SYMBOL, <span class="at">.keep_all=</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Dropping geneIDs with no available conversion</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> rna[<span class="fu">rownames</span>(rna) <span class="sc">%in%</span> geneIDs1<span class="sc">$</span>GENEID,]</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Changing the rownames to symbols</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rna) <span class="ot">&lt;-</span> geneIDs1<span class="sc">$</span>SYMBOL</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Keeping only short sample IDs</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rna) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.'</span>,<span class="st">'-'</span>,<span class="fu">substr</span>(<span class="fu">colnames</span>(rna),<span class="dv">1</span>,<span class="dv">12</span>))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="co"># write into csv</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(rna, <span class="st">'RNAseq/Tumor_HNSC_bulk_PreProcessed.csv'</span>,<span class="at">row.names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="retrieving-clinical-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Retrieving Clinical Data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Reading clinical data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>clinical <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'Clinical/TCGA_HNSC_Surival.csv'</span>, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">';'</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Adding clinical data descriptions</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Keratin state</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>clinical <span class="ot">&lt;-</span> clinical <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">K_state =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(clinical<span class="sc">$</span>K)) <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">~</span> <span class="st">"Keratin High"</span>,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(clinical<span class="sc">$</span>K)) <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">~</span> <span class="st">"Keratin Low"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#MGC Content</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>clinical <span class="ot">&lt;-</span> clinical <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">MGC_Three =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(clinical<span class="sc">$</span>MCG_total_mm2)) <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"MGC High"</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(clinical<span class="sc">$</span>MCG_total_mm2)) <span class="sc">&lt;</span> <span class="fl">0.2</span> <span class="sc">~</span> <span class="st">"MGC Low"</span>,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(clinical<span class="sc">$</span>MCG_total_mm2)) <span class="sc">&lt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(clinical<span class="sc">$</span>MCG_total_mm2)) <span class="sc">&gt;=</span> <span class="fl">0.2</span> <span class="sc">~</span> <span class="st">"MGC Medium"</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="deg-analysis-using-deseq2" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> DEG analysis using DESEq2</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Reading RNAseq data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'RNAseq/Tumor_HNSC_bulk_PreProcessed.csv'</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> <span class="fu">type.convert</span>(rna, <span class="at">as.is =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rna) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.'</span>,<span class="st">'-'</span>,<span class="fu">substr</span>(<span class="fu">colnames</span>(rna),<span class="dv">1</span>,<span class="dv">12</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Keeping only relevant bulk data</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> rna[,<span class="fu">colnames</span>(rna) <span class="sc">%in%</span> <span class="fu">rownames</span>(clinical)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Keep  only the patients with available bulk RNA seq</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>clinical_rna <span class="ot">&lt;-</span> clinical[<span class="fu">rownames</span>(clinical) <span class="sc">%in%</span> <span class="fu">colnames</span>(rna),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Need to reorder the colnames and rownames to run DESeq2</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">=</span> rna[,<span class="fu">order</span>(<span class="fu">colnames</span>(rna))]</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>clinical_rna <span class="ot">=</span> clinical_rna[<span class="fu">order</span>(<span class="fu">rownames</span>(clinical_rna)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="running-deseq2-model" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="running-deseq2-model"><span class="header-section-number">3.1</span> Running DESEq2 model</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Run DESEq2 model</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> rna,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">colData =</span> clinical_rna,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">design  =</span>  <span class="sc">~</span> MGC_Three)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(dds)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">resultsNames</span>(dds) <span class="co"># lists the coefficients</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"                        "MGC_Three_MGC.Low_vs_MGC.High"   
[3] "MGC_Three_MGC.Medium_vs_MGC.High"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>dds_vst <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">vst</span>(dds)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>matrix_vst <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(dds_vst)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Retrieving results</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">results</span>(dds, <span class="at">name=</span><span class="st">"MGC_Three_MGC.Low_vs_MGC.High"</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Change sign of Log2Fold Change</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>log2FoldChange  <span class="ot">=</span> <span class="sc">-</span>res<span class="sc">$</span>log2FoldChange</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(<span class="fu">as.data.frame</span>(res))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the DEGs into  a csv file</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res), <span class="at">file=</span><span class="st">"MGC_High_Vs_MGC_Low_DEGs.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="volcano-plot" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="volcano-plot"><span class="header-section-number">3.2</span> Volcano Plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab =</span> <span class="fu">rownames</span>(res),</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">""</span>,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="st">"MGC Low vs. MGC High"</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">selectLab =</span> <span class="fu">c</span>(<span class="st">"MARCO"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"CHIT1"</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"DCSTAMP"</span>,</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"ARG1"</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"FBP1"</span>,</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"SPP1"</span>,</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"LPL"</span>,</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"ITGAX"</span>,</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"OSCAR"</span>,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"SULT1C2"</span>),</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">xlab =</span> <span class="fu">bquote</span>(<span class="sc">~</span>Log[<span class="dv">2</span>]<span class="sc">~</span> <span class="st">"fold change"</span>),</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">ylab =</span> <span class="fu">bquote</span>(<span class="sc">~-</span>Log[<span class="dv">10</span>]<span class="sc">~</span>adjusted<span class="sc">~</span><span class="fu">italic</span>(P)),</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">pCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">FCcutoff =</span> <span class="fl">0.25</span>,</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">colAlpha =</span> <span class="dv">1</span>,</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">legendPosition =</span> <span class="st">""</span>,</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">legendIconSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">border =</span> <span class="st">"full"</span>,</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>                <span class="at">borderColour =</span> <span class="st">"black"</span>,</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>                <span class="at">gridlines.major =</span> <span class="cn">FALSE</span>,</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">gridlines.minor =</span> <span class="cn">FALSE</span>,</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>                <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">widthConnectors =</span> <span class="dv">1</span>,</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>                <span class="at">colConnectors =</span> <span class="st">'black'</span>,</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">6</span>) ,</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>                <span class="at">boxedLabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>                <span class="at">labCol =</span> <span class="st">'black'</span>,</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>                <span class="at">labFace =</span> <span class="st">'bold'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="TCGA_Analysis_for_Paper_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section>
<section id="ma-plot" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="ma-plot"><span class="header-section-number">3.3</span> MA Plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggmaplot</span>(res, <span class="at">main =</span> <span class="fu">expression</span>(<span class="st">"MGC High"</span> <span class="sc">%-&gt;%</span> <span class="st">"MGC Low"</span>),</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>   <span class="at">fdr =</span> <span class="fl">0.05</span>, <span class="at">fc =</span> <span class="fl">0.25</span>, <span class="at">size =</span> <span class="fl">0.4</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"#B31B21"</span>, <span class="st">"#1465AC"</span>, <span class="st">"darkgray"</span>),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">genenames =</span> <span class="fu">as.vector</span>(res<span class="sc">$</span>name),</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">legend =</span> <span class="st">"top"</span>, <span class="at">top =</span> <span class="dv">20</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">font.label =</span> <span class="fu">c</span>(<span class="st">"bold"</span>, <span class="dv">11</span>), <span class="at">label.rectangle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">font.legend =</span> <span class="st">"bold"</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">font.main =</span> <span class="st">"bold"</span>,</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>   <span class="at">ggtheme =</span> ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="TCGA_Analysis_for_Paper_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section>
<section id="bulk-rnaseq-signature-genes" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="bulk-rnaseq-signature-genes"><span class="header-section-number">3.4</span> Bulk RNAseq signature genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert 'res' to a data frame</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select rows with significant gene data based on criteria</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep rows where 'padj' is less than 0.05, 'padj' is not missing (not NA), and 'log2FoldChange' is greater than 0.25</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>signif_res <span class="ot">&lt;-</span> res[res<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res<span class="sc">$</span>padj) <span class="sc">&amp;</span> res<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="fl">0.25</span>, ]</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the gene names (row names) from the 'signif_res' data frame</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>signif_res <span class="ot">&lt;-</span> <span class="fu">rownames</span>(signif_res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(signif_res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "MATK"          "TENM1"         "MARCO"         "FAM184B"      
 [5] "PTGER3"        "POLB"          "NECAB2"        "ABCC1"        
 [9] "POMC"          "MMP8"          "ARG1"          "SPP1"         
[13] "PLA2G5"        "NCAN"          "CHIT1"         "TAF8"         
[17] "FAM8A1"        "FGFBP2"        "DNASE2B"       "KCNH5"        
[21] "SLC27A2"       "ITGAX"         "ZNF18"         "WIF1"         
[25] "HK3"           "ARL5A"         "PITX2"         "DCSTAMP"      
[29] "FBP1"          "HTRA4"         "OSCAR"         "NMRAL1P1"     
[33] "CYP4F11"       "FRG2C"         "LPL"           "BDNF"         
[37] "RP11-218M22.1" "PRSS36"        "KRT76"         "CYP4F3"       
[41] "MPPED1"        "ATP13A5"       "FAM69C"        "KRT77"        
[45] "SIGLEC15"      "AADACL2"       "SULT1C2"       "TMEM116"      
[49] "CES1"          "C6orf15"       "RP11-699L21.1" "LINC01272"    
[53] "RP1-223B1.1"   "AC079305.8"    "RP1-27K12.2"   "RP11-288G3.4" 
[57] "ESRRAP2"       "DUX4L26"       "LINC01010"     "RP11-431M7.3" 
[61] "LINC00942"     "RP11-20D14.6"  "RP11-21L23.4"  "CAP2P1"       
[65] "RP11-384L8.1"  "CTD-2135D7.5"  "AC005336.4"    "LINC01224"    
[69] "RP1-47A17.1"   "RP11-81A22.4" </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Write the selected gene names to a CSV file named 'signatures_genes_bulk_RNAseq.csv'</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(signif_res, <span class="st">'signatures_genes_bulk_RNAseq.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="common-gene-signature-with-visium-spots" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="common-gene-signature-with-visium-spots"><span class="header-section-number">3.5</span> Common gene signature with Visium Spots</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper function to display Venn diagram</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>display_venn <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ...){</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(VennDiagram)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.newpage</span>()</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  venn_object <span class="ot">&lt;-</span> <span class="fu">venn.diagram</span>(x, <span class="at">filename =</span> <span class="cn">NULL</span>, ...)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.draw</span>(venn_object)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>Visium_MGC_Signature <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"Visium_MGC_Signature.csv"</span>)</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>MGC_Markers <span class="ot">&lt;-</span> Visium_MGC_Signature<span class="sc">$</span>...<span class="dv">1</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">list</span>(MGC_Markers,signif_res)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Further customization</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="fu">display_venn</span>(</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">category.names =</span> <span class="fu">c</span>(<span class="st">"Visium"</span>,<span class="st">"Bulk"</span>),</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Circles</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">lwd =</span> <span class="dv">1</span>,</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"#999999"</span>, <span class="st">"#E69F00"</span>),</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Numbers</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex =</span> <span class="dv">2</span>,</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set names</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.cex =</span> <span class="dv">1</span>,</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.fontface =</span> <span class="st">"bold"</span>,</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.default.pos =</span> <span class="st">"outer"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="TCGA_Analysis_for_Paper_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="gene-ontology-based-on-degs" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="gene-ontology-based-on-degs"><span class="header-section-number">3.6</span> Gene Ontology based on DEGs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create background dataset for hypergeometric testing using all genes tested for</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co">#significance in the results</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">rownames</span>(res))</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract significant results</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>signif_res <span class="ot">&lt;-</span> res[res<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res<span class="sc">$</span>padj) <span class="sc">&amp;</span> res<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="fl">0.25</span>, ]</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>signif_genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">rownames</span>(signif_res))</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Run GO enrichment analysis</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichGO</span>(<span class="at">gene =</span> signif_genes,</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">universe =</span> all_genes,</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Hs.eg.db<span class="sc">::</span>org.Hs.eg.db,</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Output results from GO analysis to a table</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ego)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(cluster_summary,</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Description,p.adjust), <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust), <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust), <span class="at">fill =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">2.5</span>,<span class="fl">12.5</span>)) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_continuous</span>(<span class="at">low =</span> <span class="st">'royalblue'</span>, <span class="at">high =</span> <span class="st">'red4'</span>) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">'Enrichment score'</span>) <span class="sc">+</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'GO Biological processes'</span>,</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">'Enriched Terms'</span>,</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">'Cut-off lines drawn at equivalents of p=0.05, p=0.01, p=0.001'</span>) <span class="sc">+</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.001</span>)),</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="fu">c</span>(<span class="st">"dotted"</span>, <span class="st">"longdash"</span>, <span class="st">"solid"</span>),</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"black"</span>, <span class="st">"black"</span>),</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">1.5</span>, <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">24</span>) <span class="sc">+</span></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_rect</span>(),</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">16</span>, <span class="at">face =</span> <span class="st">'bold'</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">'bold'</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">'bold'</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">'bold'</span>, <span class="at">hjust =</span> <span class="fl">1.10</span>),</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">'bold'</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>,<span class="at">color =</span> cluster_summary<span class="sc">$</span>TopicNum),</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">'bold'</span>),</span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">'bold'</span>),</span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">'bold'</span>),</span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">'black'</span>),</span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Legend</span></span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key =</span> <span class="fu">element_blank</span>(), <span class="co"># removes the border</span></span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"cm"</span>), <span class="co"># Sets overall area/size of the legend</span></span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">"bold"</span>), <span class="co"># Text size</span></span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">"bold"</span>)) <span class="sc">+</span></span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-41"><a href="#cb23-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true" tabindex="-1"></a>gg1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="TCGA_Analysis_for_Paper_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#To color genes by log2 fold changes</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>signif_res_lFC <span class="ot">&lt;-</span> signif_res<span class="sc">$</span>log2FoldChange</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>clusterProfiler<span class="sc">::</span><span class="fu">cnetplot</span>(ego,</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">categorySize=</span><span class="st">"pvalue"</span>,</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">showCategory =</span> <span class="dv">12</span>,</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">foldChange=</span> signif_res_lFC,</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">vertex.label.font=</span><span class="dv">6</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="TCGA_Analysis_for_Paper_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
</section>
</section>
<section id="session-info" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Session Info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>sessioninfo<span class="sc">::</span><span class="fu">session_info</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23)
 os       macOS Big Sur ... 10.16
 system   x86_64, darwin17.0
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Europe/Paris
 date     2024-01-23
 pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version   date (UTC) lib source
 abind                  1.4-5     2016-07-21 [1] CRAN (R 4.2.0)
 annotate               1.74.0    2022-04-26 [1] Bioconductor
 AnnotationDbi          1.58.0    2022-04-26 [1] Bioconductor
 ape                    5.7-1     2023-03-13 [1] CRAN (R 4.2.0)
 aplot                  0.1.10    2023-03-08 [1] CRAN (R 4.2.0)
 backports              1.4.1     2021-12-13 [1] CRAN (R 4.2.0)
 Biobase                2.58.0    2022-11-01 [1] Bioconductor
 BiocGenerics           0.44.0    2022-11-01 [1] Bioconductor
 BiocParallel           1.32.5    2022-12-23 [1] Bioconductor
 Biostrings             2.64.1    2022-08-18 [1] Bioconductor
 bit                    4.0.5     2022-11-15 [1] CRAN (R 4.2.0)
 bit64                  4.0.5     2020-08-30 [1] CRAN (R 4.2.0)
 bitops                 1.0-7     2021-04-24 [1] CRAN (R 4.2.0)
 blob                   1.2.4     2023-03-17 [1] CRAN (R 4.2.0)
 broom                  1.0.5     2023-06-09 [1] CRAN (R 4.2.0)
 cachem                 1.0.8     2023-05-01 [1] CRAN (R 4.2.0)
 car                    3.1-2     2023-03-30 [1] CRAN (R 4.2.0)
 carData                3.0-5     2022-01-06 [1] CRAN (R 4.2.0)
 cli                    3.6.1     2023-03-23 [1] CRAN (R 4.2.0)
 clusterProfiler        4.4.4     2022-06-21 [1] Bioconductor
 codetools              0.2-19    2023-02-01 [1] CRAN (R 4.2.0)
 colorspace             2.1-0     2023-01-23 [1] CRAN (R 4.2.0)
 crayon                 1.5.2     2022-09-29 [1] CRAN (R 4.2.0)
 data.table             1.14.8    2023-02-17 [1] CRAN (R 4.2.0)
 DBI                    1.1.3     2022-06-18 [1] CRAN (R 4.2.0)
 DelayedArray           0.24.0    2022-11-01 [1] Bioconductor
 DESeq2                 1.36.0    2022-04-26 [1] Bioconductor
 digest                 0.6.32    2023-06-26 [1] CRAN (R 4.2.0)
 DO.db                  2.9       2022-10-07 [1] Bioconductor
 DOSE                   3.22.1    2022-08-30 [1] Bioconductor
 downloader             0.4       2015-07-09 [1] CRAN (R 4.2.0)
 dplyr                * 1.1.2     2023-04-20 [1] CRAN (R 4.2.0)
 EnhancedVolcano        1.14.0    2022-04-26 [1] Bioconductor
 enrichplot             1.16.2    2022-08-30 [1] Bioconductor
 evaluate               0.21      2023-05-05 [1] CRAN (R 4.2.0)
 fansi                  1.0.4     2023-01-22 [1] CRAN (R 4.2.1)
 farver                 2.1.1     2022-07-06 [1] CRAN (R 4.2.0)
 fastmap                1.1.1     2023-02-24 [1] CRAN (R 4.2.0)
 fastmatch              1.1-3     2021-07-23 [1] CRAN (R 4.2.0)
 fgsea                  1.22.0    2022-04-26 [1] Bioconductor
 formatR                1.14      2023-01-17 [1] CRAN (R 4.2.0)
 futile.logger        * 1.4.3     2016-07-10 [1] CRAN (R 4.2.0)
 futile.options         1.0.1     2018-04-20 [1] CRAN (R 4.2.0)
 genefilter             1.78.0    2022-04-26 [1] Bioconductor
 geneplotter            1.74.0    2022-04-26 [1] Bioconductor
 generics               0.1.3     2022-07-05 [1] CRAN (R 4.2.0)
 GenomeInfoDb           1.34.9    2023-02-02 [1] Bioconductor
 GenomeInfoDbData       1.2.9     2023-02-27 [1] Bioconductor
 GenomicRanges          1.50.2    2022-12-16 [1] Bioconductor
 ggforce                0.4.1     2022-10-04 [1] CRAN (R 4.2.0)
 ggfun                  0.1.1     2023-06-24 [1] CRAN (R 4.2.0)
 ggnewscale             0.4.9     2023-05-25 [1] CRAN (R 4.2.0)
 ggplot2              * 3.4.3     2023-08-14 [1] CRAN (R 4.2.1)
 ggplotify              0.1.1     2023-06-27 [1] CRAN (R 4.2.0)
 ggpubr                 0.6.0     2023-02-10 [1] CRAN (R 4.2.0)
 ggraph                 2.1.0     2022-10-09 [1] CRAN (R 4.2.1)
 ggrepel                0.9.3     2023-02-03 [1] CRAN (R 4.2.0)
 ggsignif               0.6.4     2022-10-13 [1] CRAN (R 4.2.1)
 ggtree                 3.4.4     2022-09-27 [1] Bioconductor
 glue                   1.6.2     2022-02-24 [1] CRAN (R 4.2.0)
 GO.db                  3.15.0    2022-10-07 [1] Bioconductor
 GOSemSim               2.22.0    2022-04-26 [1] Bioconductor
 graphlayouts           1.0.0     2023-05-01 [1] CRAN (R 4.2.0)
 gridExtra              2.3       2017-09-09 [1] CRAN (R 4.2.0)
 gridGraphics           0.5-1     2020-12-13 [1] CRAN (R 4.2.0)
 gtable                 0.3.4     2023-08-21 [1] CRAN (R 4.2.1)
 hms                    1.1.3     2023-03-21 [1] CRAN (R 4.2.0)
 htmltools              0.5.5     2023-03-23 [1] CRAN (R 4.2.0)
 htmlwidgets            1.6.2     2023-03-17 [1] CRAN (R 4.2.0)
 httr                   1.4.6     2023-05-08 [1] CRAN (R 4.2.0)
 igraph                 1.5.0     2023-06-16 [1] CRAN (R 4.2.0)
 IRanges                2.32.0    2022-11-01 [1] Bioconductor
 jsonlite               1.8.7     2023-06-29 [1] CRAN (R 4.2.0)
 KEGGREST               1.36.3    2022-07-14 [1] Bioconductor
 knitr                  1.43      2023-05-25 [1] CRAN (R 4.2.0)
 labeling               0.4.3     2023-08-29 [1] CRAN (R 4.2.1)
 lambda.r               1.2.4     2019-09-18 [1] CRAN (R 4.2.0)
 lattice                0.21-8    2023-04-05 [1] CRAN (R 4.2.0)
 lazyeval               0.2.2     2019-03-15 [1] CRAN (R 4.2.0)
 lifecycle              1.0.3     2022-10-07 [1] CRAN (R 4.2.1)
 locfit                 1.5-9.8   2023-06-11 [1] CRAN (R 4.2.0)
 magrittr               2.0.3     2022-03-30 [1] CRAN (R 4.2.0)
 MASS                   7.3-60    2023-05-04 [1] CRAN (R 4.2.0)
 Matrix                 1.5-4.1   2023-05-18 [1] CRAN (R 4.2.0)
 MatrixGenerics         1.10.0    2022-11-01 [1] Bioconductor
 matrixStats            1.0.0     2023-06-02 [1] CRAN (R 4.2.0)
 memoise                2.0.1     2021-11-26 [1] CRAN (R 4.2.0)
 munsell                0.5.0     2018-06-12 [1] CRAN (R 4.2.0)
 nlme                   3.1-162   2023-01-31 [1] CRAN (R 4.2.0)
 org.Hs.eg.db           3.15.0    2022-10-07 [1] Bioconductor
 patchwork              1.1.2     2022-08-19 [1] CRAN (R 4.2.0)
 pillar                 1.9.0     2023-03-22 [1] CRAN (R 4.2.0)
 pkgconfig              2.0.3     2019-09-22 [1] CRAN (R 4.2.0)
 plyr                   1.8.8     2022-11-11 [1] CRAN (R 4.2.0)
 png                    0.1-8     2022-11-29 [1] CRAN (R 4.2.0)
 polyclip               1.10-4    2022-10-20 [1] CRAN (R 4.2.0)
 purrr                  1.0.1     2023-01-10 [1] CRAN (R 4.2.0)
 qvalue                 2.30.0    2022-11-01 [1] Bioconductor
 R6                     2.5.1     2021-08-19 [1] CRAN (R 4.2.0)
 RColorBrewer           1.1-3     2022-04-03 [1] CRAN (R 4.2.0)
 Rcpp                   1.0.11    2023-07-06 [1] CRAN (R 4.2.1)
 RCurl                  1.98-1.12 2023-03-27 [1] CRAN (R 4.2.0)
 readr                  2.1.4     2023-02-10 [1] CRAN (R 4.2.0)
 reshape2               1.4.4     2020-04-09 [1] CRAN (R 4.2.0)
 rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.2.0)
 rmarkdown              2.23      2023-07-01 [1] CRAN (R 4.2.0)
 RSQLite                2.3.1     2023-04-03 [1] CRAN (R 4.2.0)
 rstatix                0.7.2     2023-02-01 [1] CRAN (R 4.2.0)
 rstudioapi             0.14      2022-08-22 [1] CRAN (R 4.2.0)
 S4Vectors              0.36.2    2023-02-26 [1] Bioconductor
 scales                 1.2.1     2022-08-20 [1] CRAN (R 4.2.0)
 scatterpie             0.2.1     2023-06-07 [1] CRAN (R 4.2.0)
 sessioninfo            1.2.2     2021-12-06 [1] CRAN (R 4.2.0)
 shadowtext             0.1.2     2022-04-22 [1] CRAN (R 4.2.0)
 stringi                1.7.12    2023-01-11 [1] CRAN (R 4.2.0)
 stringr                1.5.0     2022-12-02 [1] CRAN (R 4.2.0)
 SummarizedExperiment   1.28.0    2022-11-01 [1] Bioconductor
 survival               3.5-5     2023-03-12 [1] CRAN (R 4.2.0)
 tibble                 3.2.1     2023-03-20 [1] CRAN (R 4.2.0)
 tidygraph              1.2.3     2023-02-01 [1] CRAN (R 4.2.0)
 tidyr                  1.3.0     2023-01-24 [1] CRAN (R 4.2.0)
 tidyselect             1.2.0     2022-10-10 [1] CRAN (R 4.2.1)
 tidytree               0.4.2     2022-12-18 [1] CRAN (R 4.2.0)
 treeio                 1.20.2    2022-08-14 [1] Bioconductor
 tweenr                 2.0.2     2022-09-06 [1] CRAN (R 4.2.0)
 tzdb                   0.4.0     2023-05-12 [1] CRAN (R 4.2.0)
 utf8                   1.2.3     2023-01-31 [1] CRAN (R 4.2.0)
 vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.2.0)
 VennDiagram          * 1.7.3     2022-04-12 [1] CRAN (R 4.2.0)
 viridis                0.6.3     2023-05-03 [1] CRAN (R 4.2.0)
 viridisLite            0.4.2     2023-05-02 [1] CRAN (R 4.2.0)
 vroom                  1.6.3     2023-04-28 [1] CRAN (R 4.2.0)
 withr                  2.5.1     2023-09-26 [1] CRAN (R 4.2.1)
 xfun                   0.39      2023-04-20 [1] CRAN (R 4.2.0)
 XML                    3.99-0.14 2023-03-19 [1] CRAN (R 4.2.0)
 xtable                 1.8-4     2019-04-21 [1] CRAN (R 4.2.0)
 XVector                0.38.0    2022-11-01 [1] Bioconductor
 yaml                   2.3.7     2023-01-23 [1] CRAN (R 4.2.0)
 yulab.utils            0.0.6     2022-12-20 [1] CRAN (R 4.2.0)
 zlibbioc               1.44.0    2022-11-01 [1] Bioconductor

 [1] /Library/Frameworks/R.framework/Versions/4.2/Resources/library

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>